contracts/eosio.system/src/producer_pay.cpp

```diff
@@ -99,6 +111,25 @@ namespace eosiosystem {
          _gstate.perblock_bucket    += to_per_block_pay;
          _gstate.voters_bucket      += to_voters;
          _gstate.last_pervote_bucket_fill = ct;
+
+         // burn remaining tokenomic fees
+         if (!burn_state_sing.exists()) {
+            burn_state.last_burn_time = current_time_point();
+            burn_state_sing.set(burn_state, get_self());
+         }
+         auto burn_flag = (current_time_point() - burn_state.last_burn_time).count() >= (burn_state.usecs_burn_period);
+
+         auto burn_fees = current_fees.amount  - fees_to_use;
+         if(burn_flag && burn_fees > 0){
```

Comment:
We don't want to burn at a particular timestamp, we actually only want to burn if the eosio.fees account has a balance greater than 24 hours worth of inflation value, which would be exactly this value:
`inflation_24_hrs = static_cast<int64_t>( (continuous_rate * double(token_supply.amount) * double(twentyfour_hours_in_usec)) / double(useconds_per_year) );`
Then we can calculate the burn amount:
`burn_fees = max(0, (current_fees.amount - fees_to_use) - inflation_24_hrs)`
now, we burn if the burn_fees is greater than 0:
`if(burn_fees > 0) do_burn(burn_fees);`